pkg/tagger/local/tagger.go

```diff
@@ -215,8 +215,10 @@ func (t *Tagger) Stop() error {
 	return nil
 }
 
-// Tag returns tags for a given entity
-func (t *Tagger) Tag(entity string, cardinality collectors.TagCardinality) ([]string, error) {
+// TagReadOnly returns a read only list of tags for a given entity. No copy of
+// the internal data is made. The caller guarantes that no modification will be
+// made to the returned slice.
+func (t *Tagger) TagReadOnly(entity string, cardinality collectors.TagCardinality) ([]string, error) {
```

Comment:
If `TagReadOnly` is called by 2 goroutines, can a call to `TagReadOnly` modify the instance returned by another call to `TagReadOnly`?

```go
// Go routine 1
tags, _ := TagReadOnly()
// Do something with tags

// Go routine 2
TagReadOnly() // Can this call modify the previously returned instance (ie `tags`)
```

Reply:
No, we either hit the cache and return the same slice or we miss/partially-miss and will recreate on in [concatencaArray](https://github.com/DataDog/datadog-agent/blob/maxime/remove-tagger-copy/pkg/tagger/local/tagger.go#L277) and [reset the cache correctly](https://github.com/DataDog/datadog-agent/blob/maxime/remove-tagger-copy/pkg/tagger/local/tagger.go#L264).

Did I understood your question correctly ?

Reply:
Yes, it was exactly my question.